### PR TITLE
[specifications] update section on signature checking in move adapter docs

### PR DESCRIPTION
**Pull request by emmazzz**
## Motivation

This PR updates the section on signature checking in Move adapter specification to include more details on verifying multi-agent txns.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

Yes.

### CODE REVIEW

File: specifications/move_adapter/README.md [hunk 1]
```diff
@@ -197,17 +197,60 @@ apply to all transactions and others are specific to the type of payload.
...
+    },
+    /// Multi-agent transaction.
+    MultiAgent {
+        sender: AccountAuthenticator,
+        secondary_signer_addresses: Vec<AccountAddress>,
+        secondary_signers: Vec<AccountAuthenticator>,
+    },
+}
+
+pub enum RawTransactionWithData {
```
[the remark this answers is not in the document]
Comment (bob-wilson) [reply specifications/move_adapter/README.md:227]: To be clear: I really like the addition of the `TransactionAuthenticator` enum here, so please keep that part for sure.

**Review by bob-wilson — CHANGES_REQUESTED** [review]
Besides the comments, it seems this is still missing something about the multi-agent prologue function.

File: specifications/move_adapter/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (emmazzz) [reply specifications/move_adapter/README.md:227]: >  If `RawTransactionWithData` is not used directly in public APIs for the adapter, then it seems better not to mention it. I don't see anything else in this document that mentions it. Is that the case?
I see. That's right. `RawTransactionWithData` is not present in the public interfaces(`new`, `sign`, `verify`, etc). It is only a data structure that the signers should sign over. I will try to rephrase the text without mentioning `RawTransactionWithData`.

[the remark this answers is not in the document]
Comment (bob-wilson) [reply specifications/move_adapter/README.md]: This part looks good now. Thanks!

**Comment by bob-wilson** [comment]
> Besides the comments, it seems this is still missing something about the multi-agent prologue function.

@emmazzz Did you see this comment?

**Comment by emmazzz** [comment]
> > Besides the comments, it seems this is still missing something about the multi-agent prologue function.
> 
> @emmazzz Did you see this comment?

Ah I only looked at comments in the code and missed this one. Yes the description about multi-agent checks in the prologue in this doc is out-dated. I will add a bullet point in `Prologue Checks` mentioning the newly added prologue function.

**Review by bob-wilson — APPROVED** [review]
Looks good. Thanks!

**Pull request closed without merging**